test/index.spec.js

```diff
@@ -84,4 +84,23 @@ describe('immutableStateInvariantMiddleware', () => {
       dispatch({type: 'SOME_ACTION', x});
     }).toNotThrow();
   });
+
+  it('respects "ignore" option', () => {
+    const middlewareIgnore = (next) => {
```

Comment:
Maybe you can modify the `middleware` function at the top of this file to accept an `options` object and return a function that accepts `next`? Something like this:

```js
function middleware(options) {
  return immutableStateInvariantMiddleware()({getState});
}
```

and then modify the usages to be like:

```js
const dispatch = middleware()(next);

// and here, using the ignore option:
const dispatch = middleware({ ignore: ['foo.bar'] })(next);
```